components/markdown/src/markdown.rs

```diff
@@ -793,15 +796,50 @@ pub fn markdown_to_html(
             .position(|e| matches!(e, Event::Html(CowStr::Borrowed(CONTINUE_READING))))
             .unwrap_or(events.len());
 
+        // determine closing tags missing from summary
+        let mut tags = Vec::new();
+        for event in &events[..continue_reading] {
+            match event {
+                Event::Start(Tag::HtmlBlock) | Event::End(TagEnd::HtmlBlock) => (),
+                Event::Start(tag) => tags.push(tag.to_end()),
+                Event::End(tag) => {
+                    tags.truncate(tags.iter().rposition(|t| *t == *tag).unwrap_or(0));
+                }
+                _ => (),
+            }
+        }
+
         let mut events = events.into_iter();
 
         // emit everything up to summary
         cmark::html::push_html(&mut html, events.by_ref().take(continue_reading));
 
         if has_summary {
             // remove footnotes
-            let summary_html = FOOTNOTES_RE.replace_all(&html, "").into_owned();
-            summary = Some(summary_html)
+            let mut summary_html = FOOTNOTES_RE.replace_all(&html, "").into_owned();
+
+            // truncate trailing whitespace
+            summary_html.truncate(summary_html.trim_end().len());
+
+            // add cutoff template
+            if !tags.is_empty() {
+                let mut c = tera::Context::new();
+                c.insert("summary", &summary_html);
+                c.insert("lang", &context.lang);
+                let summary_cutoff = utils::templates::render_template(
+                    SUMMARY_CUTOFF_TEMPLATE,
+                    &context.tera,
+                    c,
+                    &None,
+                )
+                .context("Failed to render summary cutoff template")?;
+                summary_html.push_str(&summary_cutoff);
```

Comment:
Do we want a template or a string in the config? Template is more powerful but might be overkill?

Reply:
I feel like the template is ideal, although it's unlikely many people will change the template. The original idea was proposed as an easy option since even `anchor-link.html` has a simple template associated with it.

Without a template, you can't choose whether to emit an ellipsis based upon the closing punctuation, so, I'm inclined to say that the template is strictly better.